app/questionnaire_state/relationship_state_question.py

```diff
@@ -11,13 +12,34 @@ def __init__(self, id, schema_item):
     def build_repeating_state(self, user_input):
         template_answer = self.answers.pop()
         group_instance = template_answer.group_instance
-        household_answers = get_answer_store(current_user).filter(answer_id='household-full-name')
-        sorted(household_answers, key=lambda household_answer: household_answer['answer_instance'])
-        remaining_answers = household_answers[group_instance + 1:] if self.group_instance < len(household_answers) else []
+        first_names = get_answer_store(current_user).filter(answer_id='first-name')
+        last_names = get_answer_store(current_user).filter(answer_id='last-name')
 
-        for index, answer in enumerate(remaining_answers):
+        household_answers = list(zip(first_names, last_names))
+
+        household_members = []
+        for household_answer in household_answers:
+            household_members.append({
+                'first-name': household_answer[0]['value'],
+                'last-name': household_answer[1]['value'],
+            })
+
+        remaining_people = household_members[group_instance + 1:] if self.group_instance < len(household_answers) else []
+
```

Comment:
I think we can make this clearer with:
```
first_name_answers = get_answer_store(current_user).filter(answer_id='first-name')
last_name_answers = get_answer_store(current_user).filter(answer_id='last-name')

first_names = [d['value'] for d in first_name_answers]
last_names = [d['value'] for d in last_name_answers]

household_members = []
for first_name, last_name in zip(first_names, last_names):
     household_members.append({
         'first-name': first_name,
         'last-name': last_name,
      })

remaining_people = household_members[group_instance + 1:] if self.group_instance < len(household_members) else []
```

Also, zip stops after the shortest iterator finishes, so we may need to look at using itertools.zip_longest (see https://docs.python.org/3/library/functions.html#zip)